schema/functions.go

```diff
@@ -0,0 +1,51 @@
+package schema
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/hcl-lang/schema"
+
+	funcs_v0_12 "github.com/hashicorp/terraform-schema/internal/funcs/0.12"
+	funcs_v0_13 "github.com/hashicorp/terraform-schema/internal/funcs/0.13"
+	funcs_v0_14 "github.com/hashicorp/terraform-schema/internal/funcs/0.14"
+	funcs_v0_15 "github.com/hashicorp/terraform-schema/internal/funcs/0.15"
+	funcs_v1_3 "github.com/hashicorp/terraform-schema/internal/funcs/1.3"
+	funcs_generated "github.com/hashicorp/terraform-schema/internal/funcs/generated"
+)
+
+func FunctionsForVersion(v *version.Version) (map[string]schema.FunctionSignature, error) {
+	ver, err := semVer(v)
```

Comment:
I _think_ the `semVer` predates the [`Core()`](https://pkg.go.dev/github.com/hashicorp/go-version#Version.Core) method upstream, which could now be used directly and we could get rid of `semVer()`.